### PR TITLE
Feature/sprint121023/#862kft5mu Customer deposit functionality

### PR DESCRIPTION
**Pull request by bulgarchik**

### CODE REVIEW

File: FastBank.ConsoleApplication/MenuOptions.cs [hunk 1]
```diff
@@ -162,22 +163,40 @@ static public void OpenCustomerMenu()
             {
                 return;
             }
+
+            var bankAccountService = new BankAccountService();
+            var customerBankAccount = bankAccountService.GetBankAccount(ActiveCustomer);
+
+            if (customerBankAccount == null || customerBankAccount.Amount == 0)
```
Comment (AngelovGrowth) [inline FastBank.ConsoleApplication/MenuOptions.cs:170]: How user can make a deposit if customerBankAccount is null? Also, user shouldn't be required to make a deposit. Making deposit is optional, not required.

File: FastBank.Services/BankAccountService/BankAccountService.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            bankAccountRepository = new BankAccountRepository();
+        }
+        public BankAccount? GetBankAccount(Customer customer)
+        {
+            return bankAccountRepository.GetBankAccountByCustomer(customer);
+        }
+
+        public void Add(Customer customer, decimal amount)
+        {
+            if (GetBankAccount(customer) == null)
```
Comment (AngelovGrowth) [inline FastBank.Services/BankAccountService/BankAccountService.cs:22]: GetBankAccount(customer) save the result from that method in a variable and use it in the if statement

File: FastBank.Services/BankAccountService/BankAccountService.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                {
+                    Add(customer, depositAmount);
+                }
+                else
+                {
+                    customerBankAccount.DepositAmount(depositAmount);
+                    Update(customerBankAccount);
+                }
+            }
+        }
```
Comment (AngelovGrowth) [inline FastBank.Services/BankAccountService/BankAccountService.cs:70]: Code should be refactored. We can discuss it in appopriate time.

File: FastBank.ConsoleApplication/MenuOptions.cs (hunk 1 above)
Comment (bulgarchik) [reply FastBank.ConsoleApplication/MenuOptions.cs:170]: . Всеки нов клиент **трябва** да депозира сума която ще внесе в банката (това трябва да се направи след успешна регистрация на клиента.

File: FastBank.Services/BankAccountService/BankAccountService.cs (hunk 3 above)
Comment (bulgarchik) [reply FastBank.Services/BankAccountService/BankAccountService.cs:72]: image

**Pull request merged**